### PR TITLE
CAMEL-19009: camel-gson - Date Format with GsonBuilder

### PR DESCRIPTION
**Pull request by saravanakumar1987**
# Description

PR for https://issues.apache.org/jira/browse/CAMEL-19009

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Review by davsclaus — APPROVED** [review]

File: components/camel-gson/src/test/resources/org/apache/camel/component/gson/SpringGsonJsonDataFormatTest.xml [hunk 1]
```diff
@@ -56,6 +57,17 @@
             <to uri="mock:reversePojo"/>
         </route>
 
+        <route>
+            <from uri="direct:inFormatDate"/>
+            <marshal><custom ref="format_date"/></marshal>
+        </route>
+		
+		<route>
```
Comment (davsclaus) [inline components/camel-gson/src/test/resources/org/apache/camel/component/gson/SpringGsonJsonDataFormatTest.xml:65]: Would be nice to format this XML correctly

File: core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java [hunk 2]
```diff
@@ -601,6 +614,16 @@ public Builder contentTypeHeader(boolean contentTypeHeader) {
             return this;
         }
 
+        public Builder dateFormatPattern(String dateFormatPattern) {
```
Comment (davsclaus) [inline core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java:617]: Can you add javadoc with: To configure the date format while marshall or unmarshall Date fields in JSON using Gson

to the builder methods

Comment (davsclaus) [reply core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java:617]: And frankly we likely forgot to do that on other builder methods. If you spot those, you are welcome to send another PR to update those.

**Review by oscerd — APPROVED** [review]
We probably need to go through all the Json data format

File: components/camel-gson/src/test/resources/org/apache/camel/component/gson/SpringGsonJsonDataFormatTest.xml (hunk 1 above)
Comment (saravanakumar1987) [reply components/camel-gson/src/test/resources/org/apache/camel/component/gson/SpringGsonJsonDataFormatTest.xml]: Addressed

File: core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java (hunk 2 above)
Comment (saravanakumar1987) [reply core/camel-core-model/src/main/java/org/apache/camel/model/dataformat/JsonDataFormat.java:628]: Addressed

**Review by davsclaus — APPROVED** [review]

**Pull request merged**